data_model/src/account.rs

```diff
@@ -47,230 +46,94 @@ pub type AccountsMap = btree_map::BTreeMap<<Account as Identifiable>::Id, Accoun
+    #[display(fmt = "{name}@{domain_id}")]
+    #[getset(get = "pub")]
+    #[ffi_type]
+    pub struct Id {
+        /// [`Account`]'s name.
+        pub name: Name,
+        /// [`Account`]'s [`Domain`](`crate::domain::Domain`) id.
+        pub domain_id: <Domain as Identifiable>::Id,
     }
-}
 
-/// Default signature condition check for accounts.
-/// Returns true if any of the signatories have signed the transaction.
-impl Default for SignatureCheckCondition {
-    #[inline]
-    #[allow(clippy::expect_used)]
-    fn default() -> Self {
-        Self(
-            ContainsAny::new(
-                EvaluatesTo::new_unchecked(
-                    ContextValue::new(
-                        Name::from_str(TRANSACTION_SIGNATORIES_VALUE)
-                            .expect("TRANSACTION_SIGNATORIES_VALUE should be valid."),
-                    )
-                    .into(),
-                ),
-                EvaluatesTo::new_unchecked(
-                    ContextValue::new(
-                        Name::from_str(ACCOUNT_SIGNATORIES_VALUE)
-                            .expect("ACCOUNT_SIGNATORIES_VALUE should be valid."),
-                    )
-                    .into(),
-                ),
-            )
-            .into(),
-        )
+    /// Account entity is an authority which is used to execute `Iroha Special Instructions`.
+    #[derive(Debug, Display, Clone, IdEqOrdHash, Getters, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    #[allow(clippy::multiple_inherent_impl)]
+    #[display(fmt = "({id})")] // TODO: Add more?
+    #[ffi_type]
+    pub struct Account {
+        /// An Identification of the [`Account`].
+        id: Id,
+        /// Assets in this [`Account`].
+        pub assets: AssetsMap,
+        /// [`Account`]'s signatories.
+        pub signatories: Signatories,
+        /// Condition which checks if the account has the right signatures.
+        #[getset(get = "pub")]
+        pub signature_check_condition: SignatureCheckCondition,
+        /// Metadata of this account as a key-value store.
+        pub metadata: Metadata,
+        /// Roles of this account, they are tags for sets of permissions stored in `World`.
+        pub roles: RoleIds,
     }
-}
 
-declare_item! {
     /// Builder which should be submitted in a transaction to create a new [`Account`]
-    #[derive(
-        Debug,
-        Display,
-        Clone,
-        IdOrdEqHash,
-        Decode,
-        Encode,
-        Deserialize,
-        Serialize,
-        FfiType,
-        IntoSchema,
-    )]
+    #[derive(Debug, Display, Clone, IdEqOrdHash, Decode, Encode, Deserialize, Serialize, IntoSchema)]
     #[display(fmt = "[{id}]")]
+    #[ffi_type]
     pub struct NewAccount {
         /// Identification
-        pub id: <Account as Identifiable>::Id,
+        id: <Account as Identifiable>::Id,
```

Comment:
Is there a reason to make it private? We lose the ability to easily disassemble it into constituent fields without cloning

Reply:
Same question goes for other `id`s

Reply:
I guess that's a valid point, I'll make them public. I left them hidden since `IdEqOrdHash` provides the `id` method.